portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,25 +34,45 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> comments = new ArrayList<>();
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = getUserComment(request);
-    comments.add(comment);
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", comment);
+    commentEntity.setProperty("timestamp", timestamp);
```

Comment:
For these strings, you can declare constants so that they're easier to read and re-use.

In your class but outside any functions, aka at line 36/37, you can declare constants for "Comment", "text", and "timestamp"

Example: `private static final String COMMENT_PARAMETER = "Comment"`

Then, you can use your constants within your code instead of hardcoded strings, eg. `Entity commentEntity = new Entity("Comment");`